Sources/SnapshotTesting/Strategy/UIViewController.swift

```diff
@@ -28,4 +54,166 @@ extension Strategy where Snapshottable == UIViewController, Format == String {
 extension UIViewController: DefaultSnapshottable {
   public static let defaultStrategy: Strategy<UIViewController, UIImage> = .image
 }
+
+#if os(iOS)
+private func traitController(
+  for viewController: UIViewController,
+  size: CGSize,
+  traits: UITraitCollection = .init())
+  -> UIViewController
+{
+  
+  let parent = UIViewController()
+  parent.view.backgroundColor = .white
+  parent.view.frame.size = size
+  parent.preferredContentSize = parent.view.frame.size
+  parent.addChild(viewController)
+  parent.view.addSubview(viewController.view)
+  parent.setOverrideTraitCollection(traits, forChild: viewController)
+
+  viewController.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+  viewController.view.frame = parent.view.frame
+
+  parent.beginAppearanceTransition(true, animated: false)
+  parent.endAppearanceTransition()
+
+  return parent
+}
+
+public struct Environment {
+  fileprivate let device: Device
+  fileprivate let orientation: Orientation
+
+  public static let iPhoneSe = Environment(device: .iPhoneSe, orientation: .portrait)
+
+  public static func iPhoneSe(_ orientation: Orientation) -> Environment {
+    return Environment(device: .iPhoneSe, orientation: orientation)
+  }
+
+  public static let iPhone8 = Environment(device: .iPhone8, orientation: .portrait)
+
+  public static func iPhone8(_ orientation: Orientation) -> Environment {
+    return Environment(device: .iPhone8, orientation: orientation)
+  }
+
+  public static let iPhone8Plus = Environment(device: .iPhone8Plus, orientation: .portrait)
+
+  public static func iPhone8Plus(_ orientation: Orientation) -> Environment {
+    return Environment(device: .iPhone8Plus, orientation: orientation)
+  }
+
+  public static let iPadMini = Environment(device: .iPadMini, orientation: .landscape)
+
+  public static func iPadMini(_ orientation: Orientation) -> Environment {
+    return Environment(device: .iPadMini, orientation: orientation)
+  }
+
+  public static let iPadPro10_5 = Environment(device: .iPadPro10_5, orientation: .landscape)
+
+  public static func iPadPro10_5(_ orientation: Orientation) -> Environment {
+    return Environment(device: .iPadPro10_5, orientation: orientation)
+  }
+
+  public static let iPadPro12_9 = Environment(device: .iPadPro12_9, orientation: .landscape)
+
+  public static func iPadPro12_9(_ orientation: Orientation) -> Environment {
+    return Environment(device: .iPadPro12_9, orientation: orientation)
+  }
+}
+
+private enum Device {
+  case iPhoneSe
```

Comment:
what about an alternate design of statics vars/funcs on `UITraitCollection` for the various device configurations? think that could work?

Reply:
I thought about that but what about size?

Reply:
maybe `Environment` is `struct { traits: [UITraitCollection], size: CGSize }` with statics? then people could make their own, like a tv or iPhone X?

Reply:
also could we make a notch pullback? 😛 `notchramap`

Reply:
Ah cool, I like that! Not sure what's required to make iPhone X really work since the safe area insets are different. I've tried before but it wasn't quite right.